### PR TITLE
Sponge fix

### PR DESCRIPTION
**Pull request by smarras79**
SPONGE fix:
 * The sponge was not relaxing to the background horizontal velocity but to zero instead.
 * Added three more arguments to source.jl - > sponge -> Rayleigh: u_ref, v_ref, w_ref.
 * Corrected the call to Rayleigh in exemples/DGmethods/ex_001*.jl to pass u_ref, v_ref, w_ref.

### CODE REVIEW

File: src/Atmos/Model/source.jl [hunk 1]
```diff
@@ -59,7 +59,12 @@ struct RayleighSponge{FT} <: Source
   zsponge::FT
   "Sponge Strength 0 ⩽ c_sponge ⩽ 1"
   c_sponge::FT
+  #"Reference relaxation velocity"
```
Comment (asraero) [inline src/Atmos/Model/source.jl]: Suggest updating the the docs  to `are relaxed to reference values (user defined) at the top boundary.` in line `53`.

File: src/Atmos/Model/source.jl [hunk 2]
```diff
@@ -59,7 +59,12 @@ struct RayleighSponge{FT} <: Source
...
+  u_ref::FT
```
Comment (asraero) [inline src/Atmos/Model/source.jl]:
```suggestion
 "Lateral reference velocity, u[m/s]"
  u_ref::FT
```
Likewise for `v`, `w`

File: src/Atmos/Model/source.jl (hunk 1 above)
Comment (asraero) [inline src/Atmos/Model/source.jl]:
```suggestion
 "Lateral reference relaxation velocity, u [m/s]"
  u_ref::FT
```
Likewise for `v`, `w`

File: examples/DGmethods/ex_001_dycoms.jl [hunk 3]
```diff
@@ -58,81 +58,96 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
...
+function run(mpicomm, ArrayType, dim, topl, N, timeend, FT, dt, C_smag, LHF, SHF, C_drag, zmax, zsponge)
 
   grid = DiscontinuousSpectralElementGrid(topl,
-                                          FloatType = DT,
+                                          FloatType = FT,
                                           DeviceArray = ArrayType,
                                           polynomialorder = N,
-                                         )
+                                          )
+  u_ref =  7.0
```
Comment (asraero) [inline examples/DGmethods/ex_001_dycoms.jl]:
```suggestion
  u_ref =  FT(7)
```
likewise for `v_ref`, `w_ref`

File: src/Atmos/Model/source.jl [hunk 4]
```diff
@@ -68,5 +73,8 @@ function atmos_source!(s::RayleighSponge, m::AtmosModel, source::Vars, state::Va
     coeff_top = s.c_sponge * (sinpi(FT(1/2)*(z - s.zsponge)/(s.zmax-s.zsponge)))^FT(4)
     coeff = min(coeff_top, 1.0)
   end
-  source.ρu -= state.ρu * coeff
-end
+
+  u = state.ρu / state.ρ
+  u_relax_state = SVector(s.u_ref, s.v_ref, s.w_ref)
```
Comment (asraero) [inline src/Atmos/Model/source.jl]:
```suggestion
  u_relaxed = SVector(s.u_ref, s.v_ref, s.w_ref)
```
Would it make sense to reserve name `state` for either the state variables or thermo `state` ?

File: src/Atmos/Model/source.jl [hunk 5]
```diff
@@ -68,5 +73,8 @@ function atmos_source!(s::RayleighSponge, m::AtmosModel, source::Vars, state::Va
...
+  source.ρu -= state.ρ * coeff * (u - u_relax_state)
```
Comment (asraero) [inline src/Atmos/Model/source.jl]:
```suggestion
  source.ρu -= state.ρ * coeff * (u - u_relaxed)    
```
w.r.t. comment on line `78`

**Comment by asraero** [comment]
bors try

File: src/Atmos/Model/source.jl [hunk 6]
```diff
@@ -68,5 +78,8 @@ function atmos_source!(s::RayleighSponge, m::AtmosModel, source::Vars, state::Va
     coeff_top = s.c_sponge * (sinpi(FT(1/2)*(z - s.zsponge)/(s.zmax-s.zsponge)))^FT(4)
     coeff = min(coeff_top, 1.0)
```
Comment (asraero) [inline src/Atmos/Model/source.jl]:
```suggestion
    coeff = min(coeff_top, FT(1))
```

File: src/Atmos/Model/source.jl [hunk 7]
```diff
@@ -59,7 +61,15 @@ struct RayleighSponge{FT} <: Source
   zsponge::FT
   "Sponge Strength 0 ⩽ c_sponge ⩽ 1"
   c_sponge::FT
+  "Lateral reference relaxation velocity, u [m/s]"
+  u_ref::FT
```
Comment (simonbyrne) [inline src/Atmos/Model/source.jl]: could just make this a static array? 
```suggestion
  u_ref::SVector{3,FT}
```

**Comment by smarras79** [comment]
Closing this to open a new one with current code.

**Pull request closed without merging**